### PR TITLE
First Windows Forms app C#/Camera

### PR DESCRIPTION
**Pull request by AArlauskas**
Created the first Windows Form that uses AForge library to implement Camera. Able to start and stop the recording, also save the photo.
Notes:
Before using(Windows):Search bar->Camera Privacy Settings->Turn on "Allow apps to access your camera";
The saved image is always in it's default resolution quality despite the resolution changes. Needed for later use when scanning the photo.
When saving the photo adding the format is not necessary.
To run the program, you need only the files in the debug folder.

### CODE REVIEW

File: Moodie/moodie/moodie/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        private FilterInfoCollection webcam;
+        private VideoCaptureDevice cam;
+
+        private void Form1_Load(object sender, EventArgs e)
+        {
+            webcam = new FilterInfoCollection(FilterCategory.VideoInputDevice);
+            foreach(FilterInfo VideoCaptureDevice in webcam)
+            {
+                comboBox1.Items.Add(VideoCaptureDevice.Name);
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:30]: Name comboBox1 to something more obvious and definite

File: Moodie/moodie/moodie/Form1.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+            foreach (VideoCapabilities option in cam.VideoCapabilities)
+            {
+                string temp = option.FrameSize.Width.ToString() + " * " + option.FrameSize.Height.ToString();
+                comboBox2.Items.Add(temp);
+            }
+
+            comboBox2.SelectedIndex = 0;
+
+
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:44]: Unnecessary empty line here. Delete it

File: Moodie/moodie/moodie/Form1.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        private void Button1_Click(object sender, EventArgs e)
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:47]: Bad naming. Name this button accordingly to what it does. If it let's say is used for capturing photos, then its name should be `captureButton` or something else

File: Moodie/moodie/moodie/Form1.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        {
+
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:49]: Empty line

File: Moodie/moodie/moodie/Form1.cs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+        }
+
+        private void Button1_Click(object sender, EventArgs e)
+        {
+            
+            cam.NewFrame += new NewFrameEventHandler(cam_newFrame);
+            cam.Start();
+
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:52]: Empty line

File: Moodie/moodie/moodie/Form1.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        void cam_newFrame(object sender, NewFrameEventArgs eventArgs)
+        {
+            Bitmap bit = (Bitmap)eventArgs.Frame.Clone();
+            pictureBox1.Image = bit;
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:58]: Bad naming

File: Moodie/moodie/moodie/Form1.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        private void Button3_Click(object sender, EventArgs e)
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:61]: Bad naming

File: Moodie/moodie/moodie/Form1.cs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        {
+            if(cam.IsRunning)
+            {
+                cam.Stop();
+            }
+        }
+
+        private void Button2_Click(object sender, EventArgs e)
+        {
+            saveFileDialog1.InitialDirectory = Environment.CurrentDirectory;
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:71]: Bad naming

File: Moodie/moodie/moodie/Form1.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+            saveFileDialog1.FileName = "Image.jpeg";
+
+            try
+            {
+                if (saveFileDialog1.ShowDialog() == DialogResult.OK)
+                {
+                    pictureBox1.Image.Save(saveFileDialog1.FileName);
+
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:79]: Empty line

File: Moodie/moodie/moodie/Form1.cs [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+                }
+            }
+
+            catch
+            {
+                MessageBox.Show("You must first turn on the camera!");
+            }
+
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:87]: Empty line

File: Moodie/moodie/moodie/Form1.cs [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                if (saveFileDialog1.ShowDialog() == DialogResult.OK)
+                {
+                    pictureBox1.Image.Save(saveFileDialog1.FileName);
+
+                }
+            }
+
+            catch
+            {
+                MessageBox.Show("You must first turn on the camera!");
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:85]: Move this string to const. At the moment that const can be defined in the same class, but in the future we should move all string messages, alerts, etc. to a separate file

File: Moodie/moodie/moodie/Form1.cs [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            
+        }
+
+        private void Label2_Click(object sender, EventArgs e)
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Form1.cs:91]: Unnecessary and empty method

File: Moodie/moodie/moodie/Program.cs [hunk 13]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace moodie
```
Comment (tmaciulis22) [inline Moodie/moodie/moodie/Program.cs:7]: Isn't our app named Moodis?

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]

**Comment by tmaciulis22** [comment]
All in all everything looks good. Just minor code style and naming changes

File: Moodie/moodie/moodie/Form1.cs [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using AForge.Video;
+using AForge.Video.DirectShow;
+
+namespace moodie
+{
+    public partial class Form1 : Form
```
Comment (Pilkas-git) [inline Moodie/moodie/moodie/Form1.cs:15]: Change class name from Form1 to something more meaningful.

File: Moodie/moodie/moodie/Form1.cs (hunk 1 above)
Comment (Pilkas-git) [reply Moodie/moodie/moodie/Form1.cs:30]: combo box should be named "cmb + string". Naming prefixes: https://en.wikibooks.org/wiki/C_Sharp_Programming/Naming

File: Moodie/moodie/moodie/Form1.cs (hunk 3 above)
Comment (Pilkas-git) [reply Moodie/moodie/moodie/Form1.cs:47]: Change Button1 to btnCapture.

**Review by Pilkas-git — CHANGES_REQUESTED** [review]
I think that we only need the .dll  .resx and .css files.

**Pull request closed without merging**